src/libraries/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEnumerator.cs

```diff
@@ -38,6 +40,8 @@ internal FileSystemEnumerator(string directory, bool isNormalized, EnumerationOp
             _rootDirectory = Path.TrimEndingDirectorySeparator(path);
             _options = options ?? EnumerationOptions.Default;
 
+            _remainingDepth = _options.MaxRecursionDepth < 0 ? EnumerationOptions.Default.MaxRecursionDepth : _options.MaxRecursionDepth;
```

Comment:
I wonder if we should throw for negatives rather than take it as "use max value".

Reply:
It seems strange to me that we define it as Int and then throw if negative. If we only want non-negative values ​​then we should either be UInt or just use the default for them.

Reply:
One reason for using signed over unsigned is CLS-compliance. Another one is that signed integers are already the standard across many .NET APIs e.g:

[`FileStream.SetLength(long value)`](https://source.dot.net/#System.Private.CoreLib/FileStream.cs,212c0536642e78d6,references)
[`List<T>.Capacity`](https://source.dot.net/#System.Private.CoreLib/List.cs,9808f1f5ef16c436,references)

Notice that the APIs I linked guard against negatives. We should do the same here.

Reply:
Thanks for samples!  I added a value check in MaxRecursionDepth.

Done.